src/screens/auth/VerifyEmail/VerifyEmail.tsx

```diff
@@ -4,26 +4,50 @@ import { View } from 'react-native';
 import { Text } from 'src/components';
 import { ScreenTemplate } from 'src/components/templates';
 import { RootStackParamList } from 'src/navigation';
+import { useAppDispatch } from 'src/store';
+import { AsyncThunks } from 'src/store/thunks';
 
 import styles from './VerifyEmail.style';
 
-const VerifyEmail = () => {
+const VerifyEmail = ({ route }: any) => {
   const navigation = useNavigation<NavigationProp<RootStackParamList>>();
+  const dispatch = useAppDispatch();
+  const [optionalMessage, setOptionalMessage] = React.useState('Waiting...');
+
+  const verification = async () => {
```

Comment:
Use [`useCallback`](https://react.dev/reference/react/useCallback) here to avoid potential unnecessary rerenders here